src/common/dateInput/DateInputLogic.tsx

```diff
@@ -0,0 +1,246 @@
+import React from 'react';
+import defaultTo from 'lodash/defaultTo';
+
+const DATE_INPUT_ALLOWED_CHARACTERS = [
+  '0',
+  '1',
+  '2',
+  '3',
+  '4',
+  '5',
+  '6',
+  '7',
+  '8',
+  '9',
+];
+const INPUT_CONFIGS = {
+  // We are invoking type number in order to allow platforms invoke
+  // behavior that allows more convenient  number input.
+  type: 'number',
+  // Sets the minimum allowed number as 0 for chrome controls. The user
+  // is still able to input a negative number by hand if they choose.
+  min: 0,
+  // 1) Ensures that a number pad is shown on iOS instead of the regular
+  // keyboard
+  // 2) Blocks iOS devices from inputting characters other than numbers
+  pattern: '\\d*',
+};
+
+type DateObject = {
+  date?: number;
```

Comment:
Would it be even better to use `day` or `dayofMonth`? I am quite sure `date` will cause confusion, since date is usually a specific day on a calendar: 

https://en.wikipedia.org/wiki/Calendar_date

Reply:
I agree that that's the case generally--but the developer interacting with this code should be aware of how `Date` object in JS behave, because this component is in essence a control for creating JS `Date`s. Because of that, I'd value the development context here more than the "human" (:D) context. I.e. the outward facing naming should adopt the general pattern, but the internal naming should align with the concepts that are important for the internal implementation. In that sense I think the implementation should be looked from this perspective: _I am a developer of this component_, instead of this perspective _I am a human_/_I am a consumer of this component_.

But alas it's not a big enough distinction that I'd want to put my foot down, so I'll change it.